src/lib/PageTracker.js

```diff
@@ -0,0 +1,410 @@
+     * @return {boolean}
+     */
+    isActive() {
+        return this.active;
+    }
+
+    /**
+     * Start the content insights interval, to report data each number of seconds defined on a Const
+     *
+     * @public
+     * @return {void}
+     */
+    startTracking() {
+        this.stopTracking();
+        this.startTime = Date.now();
+
+        this.reportingIntervalId = setInterval(() => {
+            this.report();
+        }, REPORTING_INTERVAL);
+    }
+
+    /**
+     * Stop the content insights interval, for example, if a user goes idle or the user changes the page
+     * and report the data
+     *
+     * @public
+     * @return {void}
+     */
+    stopTracking() {
+        if (!this.reportingIntervalId) return;
+
+        clearInterval(this.reportingIntervalId);
+        this.reportingIntervalId = null;
+        this.report();
+    }
+
+    /**
+     * Handle the page change event. At this moment, it would stop the current timer interval,
+     * report the data for the previous page, and start the interval again
+     *
+     * @public
+     * @param {Object} eventData - Event data payload received by a Page change event
+     * @return {void}
+     */
+    handleViewerPageChange(eventData) {
+        if (this.active) {
+            this.stopTracking();
+            this.setCurrentPage(eventData);
+            this.startTracking();
+        } else {
+            this.setCurrentPage(eventData);
+        }
+    }
+
+    /**
+     * Set the idle timeout to call the inactiveUserHandler when time passes
+     *
+     * @private
+     * @return {void}
+     */
+    setIdleTimer() {
+        this.idleTimerTimeoutId = setTimeout(this.inactiveUserHandler, IDLE_TIMER);
+    }
+
+    /**
+     * Clear the idle timeout
+     *
+     * @private
+     * @return {void}
+     */
+    clearIdleTimeout() {
+        clearTimeout(this.idleTimerTimeoutId);
+        this.idleTimerTimeoutId = null;
+    }
+
+    /**
+     * Reset the idle timeout to avoid calling the inactiveUserHandler if the user is active.
+     * Also, if the content insights interval is stopped (because the user was inactive) restart the interval
+     *
+     * @private
+     * @return {void}
+     */
+    resetIdleTimer() {
+        if (!this.reportingIntervalId && this.active) {
+            this.startTracking();
+        }
+        this.clearIdleTimeout();
+        this.setIdleTimer();
+    }
+
+    /**
+     * Call the resetIdleTimer if the user is active
+     *
+     * @private
+     * @return {void}
+     */
+    activeUserHandler() {
+        this.resetIdleTimer();
+    }
+
+    /**
+     * Stops the content insights interval if the user goes inactive
+     *
+     * @private
+     * @return {void}
+     */
+    inactiveUserHandler() {
+        this.stopTracking();
+    }
+
+    /**
+     * Binds DOM listeners to a container to manage if the user goes active or inactive.
+     *
+     * @private
+     * @return {void}
+     */
+    bindIdleUserEvents() {
```

Comment:
Do we need to add a listener for tab visibility changes?

Reply:
Yes, because if the viewer leaves the page to see another tab, they are not actually viewing the document, so we need to stop tracking. The same goes if the viewer comes back and we need to continue.